ime/app/src/main/java/com/anysoftkeyboard/dictionaries/SuggestionsProvider.java

```diff
@@ -264,60 +279,74 @@ public void setupSuggestionsForKeyboard(
 
-        for (DictionaryAddOnAndBuilder dictionaryBuilder : dictionaryBuilders) {
+        for (int i = 0; i < dictionaryBuilders.size(); i++) {
             try {
                 Logger.d(
                         TAG,
                         " Creating dictionary %s (%s)...",
-                        dictionaryBuilder.getId(),
-                        dictionaryBuilder.getLanguage());
-                final Dictionary dictionary = dictionaryBuilder.createDictionary();
+                        dictionaryBuilders.get(i).getId(),
+                        dictionaryBuilders.get(i).getLanguage());
+                final Dictionary dictionary = dictionaryBuilders.get(i).createDictionary();
                 mMainDictionary.add(dictionary);
                 Logger.d(
                         TAG,
                         " Loading dictionary %s (%s)...",
-                        dictionaryBuilder.getId(),
-                        dictionaryBuilder.getLanguage());
+                        dictionaryBuilders.get(i).getId(),
+                        dictionaryBuilders.get(i).getLanguage());
                 disposablesHolder.add(
                         DictionaryBackgroundLoader.loadDictionaryInBackground(cb, dictionary));
             } catch (Exception e) {
-                Logger.e(TAG, e, "Failed to create dictionary %s", dictionaryBuilder.getId());
+                Logger.e(
+                        TAG,
+                        e,
+                        "Failed to create dictionary %s",
+                        dictionaryBuilders.get(i).getId());
             }
 
             if (mUserDictionaryEnabled) {
                 final UserDictionary userDictionary =
-                        createUserDictionaryForLocale(dictionaryBuilder.getLanguage());
+                        createUserDictionaryForLocale(dictionaryBuilders.get(i).getLanguage());
                 mUserDictionary.add(userDictionary);
                 Logger.d(
-                        TAG, " Loading user dictionary for %s...", dictionaryBuilder.getLanguage());
+                        TAG,
+                        " Loading user dictionary for %s...",
+                        dictionaryBuilders.get(i).getLanguage());
                 disposablesHolder.add(
                         DictionaryBackgroundLoader.loadDictionaryInBackground(userDictionary));
                 mUserNextWordDictionary.add(userDictionary.getUserNextWordGetter());
             } else {
                 Logger.d(TAG, " User does not want user dictionary, skipping...");
             }
-
-            if (mQuickFixesEnabled) {
-                final AutoText autoText = dictionaryBuilder.createAutoText();
+            // if mQuickFixesEnabled and mQuickFixesSecondDisabled are true
+            // it  activates autotext only to the current keyboard layout language
+            if ((mQuickFixesEnabled && mQuickFixesSecondDisabled && i == 0)
```

Comment:
this could be changed to:
```
if (mQuickFixesEnabled && (i == 0 || !mQuickFixesSecondDisabled)) {
```

Reply:
I knew there should exist an easier way for that if line! Thank you